# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/build/FileLocation.h
# Commit: b07887474f51
# Full Hash: b07887474f51c5d5e01cb0188c18c8618c77be57
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-07-02 03:58:28
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1627075 - Route Omnijar requests through StartupCache r=froydnj
#   
#   This should be a relatively straightforward patch. Essentially, we implement
#   a wrapper class (and friends) around nsZipArchive (and friends), which transparently
#   caches entries from the underlying zip archive in the StartupCache. This will break
# ==============================================================================

diff -r 65c0e6790a33 -r b07887474f51 xpcom/build/FileLocation.h
--- a/xpcom/build/FileLocation.h	Wed Jul 01 17:55:38 2020 +0000
+++ b/xpcom/build/FileLocation.h	Wed Jul 01 17:09:53 2020 +0000
@@ -17,6 +17,8 @@
 
 namespace mozilla {
 
+class CacheAwareZipReader;
+
 class FileLocation {
  public:
   /**
@@ -27,7 +29,7 @@
    * - in archives within archives
    * As such, it stores a path within an archive, as well as the archive
    * path itself, or the complete file path alone when on a filesystem.
-   * When the archive is in an archive, an nsZipArchive is stored instead
+   * When the archive is in an archive, an CacheAwareZipReader is stored instead
    * of a file path.
    */
   FileLocation();
@@ -45,9 +47,14 @@
 
   /**
    * Constructors for path within an archive. The archive can be given either
-   * as nsIFile or nsZipArchive.
+   * as nsIFile or CacheAwareZipReader.
    */
-  FileLocation(nsIFile* aZip, const char* aPath);
+  FileLocation(nsIFile* aFile, const char* aPath);
+
+  /**
+   * Constructors for path within a zip archive.
+   */
+  FileLocation(CacheAwareZipReader* aZip, const char* aPath);
 
   FileLocation(nsZipArchive* aZip, const char* aPath);
 
@@ -61,10 +68,12 @@
    */
   void Init(nsIFile* aFile);
 
-  void Init(nsIFile* aZip, const char* aPath);
+  void Init(nsIFile* aFile, const char* aPath);
 
   void Init(nsZipArchive* aZip, const char* aPath);
 
+  void Init(CacheAwareZipReader* aZip, const char* aPath);
+
   /**
    * Returns an URI string corresponding to the file location
    */
@@ -78,7 +87,7 @@
    */
   already_AddRefed<nsIFile> GetBaseFile();
 
-  nsZipArchive* GetBaseZip() { return mBaseZip; }
+  CacheAwareZipReader* GetBaseZip() { return mBaseZip; }
 
   /**
    * Returns whether the "base file" (see GetBaseFile) is an archive
@@ -119,7 +128,7 @@
    protected:
     friend class FileLocation;
     nsZipItem* mItem;
-    RefPtr<nsZipArchive> mZip;
+    RefPtr<CacheAwareZipReader> mZip;
     mozilla::AutoFDClose mFd;
   };
 
@@ -131,7 +140,7 @@
 
  private:
   nsCOMPtr<nsIFile> mBaseFile;
-  RefPtr<nsZipArchive> mBaseZip;
+  RefPtr<CacheAwareZipReader> mBaseZip;
   nsCString mPath;
 }; /* class FileLocation */
 